packages/core-compat-api/src/collectLegacyRoutes.tsx

```diff
@@ -173,9 +173,8 @@ export function collectLegacyRoutes(
     (route: ReactNode) => {
       // TODO(freben): Handle feature flag and permissions framework wrapper elements
       if (!React.isValidElement(route) || route.type !== Route) {
-        return;
+        throw new Error('Invalid <Route /> element has been detected.');
```

Comment:
```suggestion
        throw new Error(`Invalid element inside FlatRoutes, expected Route but found ${route.type}.`);
```

at least in the code path where we know that it's a valid element? There's a little bit too little info in the message otherwise, to understand what went wrong and for us to help debug.